src/test/kotlin/com/charleskorn/kaml/YamlReadingTest.kt

```diff
@@ -831,6 +838,119 @@ object YamlReadingTest : Spek({
                 }
             }
 
+            val sealedYaml = Yaml(context = sealedModule)
+
+            context("given some tagged input representing an object where the resulting type should be a sealed class (int)") {
+                val input = """
+                    element: !<sealedInt>
+                        value: 3
+                """.trimIndent()
+
+                context("parsing that input") {
+                    val result = sealedYaml.parse(SealedWrapper.serializer(), input)
+                    it("deserializes it to a Kotlin object") {
+                        assert(result).toBe(SealedWrapper(TestSealedStructure.SimpleSealedInt(3)))
+                    }
+                }
+            }
+
+            context("given some tagged input representing an object where the resulting type should be a sealed class (string)") {
+                val input = """
+                    element: !<sealedString>
+                        value: "asdfg"
+                """.trimIndent()
+
+                context("parsing that input") {
+                    val result = sealedYaml.parse(SealedWrapper.serializer(), input)
+                    it("deserializes it to a Kotlin object") {
+                        assert(result).toBe(SealedWrapper(TestSealedStructure.SimpleSealedString("asdfg")))
+                    }
+                }
+            }
+
+            context("given some tagged input representing a list of objects where the resulting type should be a sealed class") {
+                val input = """
+                    - element: !<sealedString>
+                        value: "some"
+                    - element: !<sealedInt>
+                        value: -987
+                    - element: !<sealedInt>
+                        value: 654
+                    - element: !<sealedString>
+                        value: "tests"
+                """.trimIndent()
+
+                context("parsing that input") {
+                    val result = sealedYaml.parse(SealedWrapper.serializer().list, input)
+                    it("deserializes it to a Kotlin object") {
+                        assert(result).toBe(
+                            listOf(
+                                SealedWrapper(TestSealedStructure.SimpleSealedString("some")),
+                                SealedWrapper(TestSealedStructure.SimpleSealedInt(-987)),
+                                SealedWrapper(TestSealedStructure.SimpleSealedInt(654)),
+                                SealedWrapper(TestSealedStructure.SimpleSealedString("tests"))
+                            )
+                        )
+                    }
+                }
+            }
+
+            val interfaceYaml = Yaml(context = interfaceModule)
```

Comment:
Is there something special that we have to do with interfaces as compared to classes? I'm not clear if these tests are required.